### DISCUSSION TITLE
About Nginx reverse proxy DDNS DNS, cache issue

### DISCUSSION THREAD

**Discussion by lixuewang** [apache/apisix#5753] [Q&A]
在使用nginx做反向代理的，将请求发送到一个动态DDNS域名的时候，该动态DDNS域名对应的IP是A ，刚开始运行一切正常，但是当运行了一段时间以后，该动态DDNS域名对应的IP变了之后(例如对应的IP由A变为B)，nginx的转发仍然还在向原先的IP A发送请求，导致反向代理中断，此时reload nginx后才会重新恢复正常，且日志显示数据转发到新的IP B了，请问如何让nginx自动去重新解析域名，而不用每次出现问题了人工去reload？

造成这个问题的主要原因是，在Nginx启动的时候会做域名解析，然后把IP缓存起来以后会一直使用解析到的IP并且不会再更改，除非重新启动Nginx，Nginx才会重新解析域名

APISIX是如何解决这个问题的 ？

**Comment by membphis** [comment]
@lixuewang pls use English in public channel, many thx

> APISIX是如何解决这个问题的 ？

Because APISIX needs to support a complete DNS function, APISIX does not use Nginx's DNS, but uses the Lua land to resolve it separately (allowing to set a custom TTL).

So APISIX doesn't have this problem you mentioned.

**Reply by lixuewang** [reply]
my case:
systemA->apisix->(domainname)systemB(has ip IPA and ipB)

1、systemA->apisix->IPA
2、when domainname resolve ipB

apisix still has many long established link to ipA

but I wish apisix connect to IPB .because IPA is read only ,cannot write

how can I resolve this question？

**Reply by juzhiyuan** [reply]
Hi @membphis, would you please give some hints?

**Reply by membphis** [reply]
> but I wish apisix connect to IPB .because IPA is read only ,cannot write

how do you know this? 

When the DNS result expires, DNS resolution will be triggered again.
When the DNS result has not expired, the latest DNS resolution result will be reused directly.

For more details, please see dns resolution implementation: https://github.com/Kong/lua-resty-dns-client/blob/master/src/resty/dns/client.lua.

**Comment by lixuewang** [comment]
当域名解析后的IP地址发生变化后，APISIX已经建立的连接池中的长连接是无法释放的吧 ？

**Comment by lixuewang** [comment]
当域名解析后的IP地址发生变化后，APISIX已经建立的连接池中的长连接是无法立即释放的吧 ？

**Reply by membphis** [reply]
> 当域名解析后的IP地址发生变化后，APISIX已经建立的连接池中的长连接是无法立即释放的吧 ？

New requests will use the new upstream IP address.
The old IP address will be automatically released by nginx when the connection exceeds the keepalive time.

**Reply by lixuewang** [reply]
yes,you are right.

before  "the connection exceeds the keepalive time."  , the service is not available。

i have no solution to slove the problem.
